### PR TITLE
Fix Conda install on MAC OS

### PR DESCRIPTION
**Pull request by bam241**
This prevent building `libcyclus` and `cyclus` binary against PYTHONLIB on MACOS.

I believe MACOS does not do exactly the same thing as Linux based system when building static/charged libs.

Without this, when importing cyclus python packages the system ends up importing twice the PYTHONLIB (ones because it is in python environment one because it is included in libcyclus, on which cyclus python package relies on...) — strangely it only occurs when using Conda python...

I tried this on macOS (no Conda) and Ubuntu (no Conda) 
The CI will check Ubuntu (Conda) and @yarden-livnat has tested in on MACOS(Conda)

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -337,7 +337,10 @@ IF(NOT CYCLUS_DOC_ONLY)
 
         # make sure we know about having python
         add_definitions(-DCYCLUS_WITH_PYTHON)
-        set(LIBS ${LIBS} ${PYTHON_LIBRARIES})
+        IF (NOT APPLE)
+            MESSAGE("ININININHBJGVHGCVGFCGFCGFCGFCGFCGFCGFCGFC:")
```
Comment (gonuke) [inline CMakeLists.txt]: Is this necessary?? or something leftover from debugging?

Comment (bam241) [reply CMakeLists.txt]: pushy sorry, forgot to clean this up....

**Comment by bam241** [comment]
> I sure wish I understood this better, but will trust your testing experience.
> 
> Needs a news file!

I think it is related to how MacOS forms the Static lib if (I understand correctly part of the answer can be found there: https://stackoverflow.com/questions/2339679/what-are-the-differences-between-so-and-dylib-on-osx)
basically it build static lib as bundles and if you link against an other static lib it cannot be loaded as a shared lib but only as a "bundle".
This might results in loading twice the PYTHONLIB.
For a reason, I ignore (they might be built differently), when using the `macport` PYTHONLIB you don't get that issue.

Also on MACOS (regardless on the` PYTONLIB` provenance), importing `PYTHONLIB` is only required when building the binary.
Technically the `libcyclus` (C++) would need it too, but as it is only loaded thought the Python package or the binary, MACOS gets the `Python` symbols either from the Python Environment or the binary that is linked against it.
This does not work on Ubuntu, libcyclus needs to be linked against `PYTHONLIB`.

**Comment by bam241** [comment]
@gonuke just added the news file

File: CMakeLists.txt [hunk 2]
```diff
@@ -337,7 +337,9 @@ IF(NOT CYCLUS_DOC_ONLY)
 
         # make sure we know about having python
         add_definitions(-DCYCLUS_WITH_PYTHON)
-        set(LIBS ${LIBS} ${PYTHON_LIBRARIES})
+        IF (NOT APPLE)
+            set(LIBS ${LIBS} ${PYTHON_LIBRARIES})
```
Comment (gonuke) [inline CMakeLists.txt]: I guess another question - do we need to add `PYTHON_LIBRARIES` for linux.  I guess it's clear we don't need it for MacOS, but could we do without it for Linux?

Comment (gonuke) [reply CMakeLists.txt]: That is, linux may tolerate it, but it still may not be necessary??? (pure speculation...)

Comment (bam241) [reply CMakeLists.txt]: I'll try this

File: CMakeLists.txt [hunk 3]
```diff
@@ -337,7 +337,9 @@ IF(NOT CYCLUS_DOC_ONLY)
...
+        ENDIF (!APPLE)
```
Comment (nuclearkatie) [inline CMakeLists.txt]: Is there a reason you open the if statement with ```NOT APPLE``` but end it with ```!APPLE``` ? Does it matter if they exactly match?

File: CMakeLists.txt (hunk 2 above)
Comment (bam241) [reply CMakeLists.txt]: this is required for `linux`

File: CMakeLists.txt (hunk 3 above)
Comment (bam241) [reply CMakeLists.txt]: fair point fixing it

File: cli/CMakeLists.txt [hunk 4]
```diff
@@ -8,10 +8,14 @@ INCLUDE_DIRECTORIES(BEFORE ${CYCLUS_CORE_INCLUDE_DIRS})
...
     cyclus.cc
     )
+IF (APPLE)
+    SET(LIBS ${PYTHON_LIBRARIES} ${LIBS})
+ENDIF(APPLE)
 
 ADD_EXECUTABLE(cyclus_cli ${CYCLUS_SOURCE})
 set_target_properties(cyclus_cli PROPERTIES OUTPUT_NAME cyclus)
 
+
```
Comment (stompsjo) [inline cli/CMakeLists.txt]:
```suggestion
```

I don't think this extra line is needed

**Comment by bam241** [comment]
@gonuke we need you to approve the PR to be able to merge.

**Comment by gonuke** [comment]
Thanks @bam241

**Pull request merged**